packages/ui-scripts/lib/build/buildThemes/generateComponents.js

```diff
@@ -75,68 +73,69 @@ export const resolveReferences = (semantics, key) => {
   if (!isNaN(Number(value))) {
     return `${value},\n`
   }
-
   return `"${value}",\n`
 }
 
 const generateComponent = (data) => {
   const formattedSemantic = formatComponent(data)
-
   return resolveReferences(formattedSemantic)
 }
 
-const getGenericType = (value) => {
-  if (isReference(value)) {
-    return `Semantics${value
-      .slice(1, -1)
-      .split('.')
-      .map((val) => `['${val}']`)
-      .join('')}`
-  }
-  return 'string'
-}
-
 const parseType = (key, tokenObject, acc) => {
   let ret = acc
   if (tokenObject.type) {
+    // Add composition token support if needed
     switch (tokenObject.type) {
       case 'border':
-        // This type is coming from Token Studio
-        ret += '{ style: "solid" | "dashed" }'
+        ret += 'TokenBorderValue'
         break
-      case 'borderWidth':
+      // the following types are coming from SingleXYToken in @token-studio/types
+      // we could add them as imports from @token-studio/types if needed
```

Comment:
I decided to just hardcode these simple types here, because importing them all if they are required would be lots of extra code